### PR TITLE
added support for java.bazel.staticProcessStreamSocket vmarg

### PR DESCRIPTION
**Pull request by lonhutt**

### CODE REVIEW

**Review by brian-mcnamara — APPROVED** [review]

**Review by bmp4070 — APPROVED** [review]

File: src/loggingTCPServer.ts [hunk 1]
```diff
@@ -11,6 +11,16 @@ const RETRY_INTERVAL = 5000; // ms
 let server: Server | undefined;
 
 function startTCPServer(attempts = 0): Promise<number> {
+	let port = 0;
+	if (workspace.getConfiguration('java').has('jdt.ls.vmargs')) {
+		const vmargs = workspace
+			.getConfiguration('java')
+			.get<string>('jdt.ls.vmargs');
+		if (vmargs?.includes('java.bazel.staticProcessStreamSocket')) {
+			port = parseInt(vmargs.split('java.bazel.staticProcessStreamSocket=')[1]);
```
Comment (guw) [inline src/loggingTCPServer.ts:20]: Wouldn't this require splitting by space first?

Comment (lonhutt) [reply src/loggingTCPServer.ts:20]: i don't think so, if we split on the thing we're looking for we know the next value in the array is always the port number.

If we split on space first then we have to find the element with the value we care about, then split that on `=`; which feel unnecessary.

Comment (guw) [reply src/loggingTCPServer.ts:20]: Are you saying that `"-Djava.bazel.staticProcessStreamSocket=123 -Dfoo=bar -XX".split('java.bazel.staticProcessStreamSocket=')` will return `123` as a single item but not `123 -Dfoo=bar -XX`?

Comment (lonhutt) [reply src/loggingTCPServer.ts:20]: ah, yeah, my bad

**Comment by guw** [comment]
@lonhutt I think this little piece of code would be a nice candidate for a unit test. Can you please add one?

**Comment by guw** [comment]
@lonhutt I merged this one meanwhile. Can you add the test in a follup up PR please?

**Pull request merged**